code/datums/gamemode/role/time_agent.dm

```diff
@@ -0,0 +1,193 @@
+/**
+	Time agent
+
+	You've got 5 minutes of playtime to do something weird
+
+	If you succeed, the stations threat level is escalated
+
+	If you fail, something weird happens.
+
+	Are you a bad enough dude to make sure a corgi, a rubber duck, and a bucket are in the same place at the same time?
+
+**/
+
+/datum/role/time_agent
+	name = "time agent"
+	id = TIMEAGENT
+	required_pref = ROLE_MADNESS
```

Comment:
make sure you rebase for your testing, your branch is old enough that dynamic checks the ID for bans/preferences (so it would look for a preference called time agent instead of madness). I changed this with the minor roles PR so that it goes off required pref instead.